task_processing/plugins/mesos/retrying_executor.py

```diff
@@ -36,34 +36,49 @@ def event_with_retries(self, event):
         return event.transform(
             ('extensions', 'RetryingExecutor/tries'),
             "{}/{}".format(
-                1 + self.retries - self.task_retries.get(event.task_id, -1),
+                self.task_retries[event.task_id],
                 self.retries
             )
         )
 
     def retry(self, event):
-        current_retries = self.task_retries.get(event.task_id, -1)
-        if current_retries <= 0:
+        with self.task_retries_lock:
+            current_retry_attempt = self.task_retries[event.task_id]
+
+        if current_retry_attempt == self.retries:
             return False
 
         log.info(
             'Retrying task {}, {} of {}, fail event: {}'.format(
-                event.task_config.name, 1 + self.retries - current_retries,
+                event.task_config.name, current_retry_attempt,
                 self.retries, event.raw
             )
         )
 
-        self.run(event.task_config)
         with self.task_retries_lock:
-            self.task_retries = self.task_retries.update_with(
-                sub, {event.task_id: 1}
+            self.task_retries = self.task_retries.set(
+                event.task_id,
+                current_retry_attempt + 1
             )
+        self.run(event.task_config)
+
         return True
 
     def retry_loop(self):
         while True:
             while not self.src_queue.empty():
                 e = self.src_queue.get()
+                # This is to remove trailing '-retry*'
+                original_task_id = '-'.join([item for item in
+                                             e.task_id.split('-')[:-1]])
+
+                # Check if the update is for current attempt. Discard if
+                # it is not.
+                if not self._is_update_duplicate(e, original_task_id):
```

Comment:
function name doesn't really reflect the purpose, could you change to `is_current_attempt` or something?

Reply:
Sure. Fixed.